terraform/bigquery.tf

```diff
@@ -180,6 +180,105 @@ resource "google_bigquery_table_iam_member" "event_viewers" {
   member     = each.value
 }
 
+# Events table with clustering and partitioning / IAM
+resource "google_bigquery_table" "raw_events_table" {
```

Comment:
this is the same except adding partition and cluster, why not update the existing one?

Reply:
You cannot add partitioning to an existing table

Reply:
Also I changed the schema, events is now JSON rather than STRING

Reply:
Where? This is the schema for the pubsub topic: https://github.com/abcxyz/github-metrics-aggregator/pull/136/files#diff-c0e7aafb097169e05cfc80fcfa7a7379e723829aebeda7fb0c1113634a8ab785R144-R153

Reply:
....I believe this is what causes the issue when I last tried this.

Reply:
I guess, have you tested this or are you doing this to test it?

Reply:
https://cloud.google.com/pubsub/docs/bigquery#protobuf-to-zetasql

So string is the proper protobuf type to import to a JSON column. 

I haven't tested it (figure this will tell us if it will work), but we use the same pattern in PMAP: https://github.com/abcxyz/pmap/blob/main/pkg/server/pubsub.go#L64  https://github.com/abcxyz/pmap/blob/main/terraform/modules/pubsub-bigquery/main.tf#L69

Reply:
They are using a non-schema approach that has a specific format from pubsub, so I dont think this will work but happy to test it out.

[Details part 1](https://cloud.google.com/pubsub/docs/subscription-properties#bigquery:~:text=If%20you%20don%27t%20select%20the%20Use%20topic%20schema%20option%2C%20ensure%20that%20the%20BigQuery%20table%20has%20a%20column%20called%20data%20of%20type%20BYTES%2C%20STRING%2C%20or%20JSON.%20Pub/Sub%20writes%20the%20message%20to%20this%20BigQuery%20column.)
[Details part 2](https://cloud.google.com/pubsub/docs/subscription-properties#write-metadata)

Reply:
I see, it is a bit different. Only one way to find out if it works